drawing_shape2

```diff
@@ -0,0 +1,201 @@
+        qp.drawPath(path)
+       
+        qp = QPainter()
+        qp.begin(self)
+        path = QPainterPath()
+        
+                
+        ### rysowanie granicy kraju
+        pen = QPen(Qt.black, 3, Qt.SolidLine)
+        qp.setPen(pen)
+        #pętla rysująca kształt z shp - poligon, za pętlą instrukcja połączenia ostatniego punktu z pierwszym
+        path.moveTo(shap_pol[0].points[0][0]/1000,h-shap_pol[0].points[0][1]/1000)
+        for i in range (len(shap_pol[0].points)-1):
+            path.lineTo(shap_pol[0].points[i+1][0]/1000,h-shap_pol[0].points[i+1][1]/1000)
+        path.lineTo(shap_pol[0].points[0][0]/1000,h-shap_pol[0].points[0][1]/1000)
+
+        qp.drawPath(path)
+        qp.end()
+        
+        
+        ### rysowanie miast - jako elips
+    def drawPoints(self):
+        
+        h = self.h # zmienna lokalna h otrzymuje taką wartosć jaką posiada zmienna klasowa self.h
+        
+        qp = QPainter()
+        qp.begin(self)
+        
+        pen = QPen(Qt.red, 2, Qt.SolidLine)
+        qp.setPen(pen)
+        
+        for i in range(len(shap_miasta)): #pętla po liczbie miast
+            lon = (shap_miasta[i].points[0][0])/1000
+            lat = h-(shap_miasta[i].points[0][1])/1000
+            qp.drawEllipse(lon-2, lat-2, 5, 5) #rysuje okręgi o rozmiarze 5 i lewym gornym narozniku: wsp-2 (aby srodek byl tam gdzie miejscowosc) 
+        
+        qp.end()
+ 
+    """  
+    #funkcja sledzaca ruch myszy - teraz juz zbędna, bo jest mousePressEvent
+    def mouseMoveEvent(self, e): # sledzenie myszy
+        
+        x = e.x()
+        y = e.y()
+        
+        text = "x: {0},  y: {1}".format(x, y)
+        self.label.setText(text)
+    """    
+    def mousePressEvent (self, ev):
+        """
+        funkcja pobierająca wspołrzędne kliknięcia i wyswietlająca je
+        """
+        if ev.button() == 1:
+            
+            x = ev.x()
+            y = ev.y()
+        
+            text = "x: {0},  y: {1}".format(x, y)
+            self.label.setText(text)
+        #proba narysowania symbolu w miejscu kliknięcia
+        """
+        qp = QPainter()
+        qp.begin(self)
+        
+        pen = QPen(Qt.red, 3, Qt.SolidLine)
+        qp.setPen(pen)
+        qp.drawEllipse(x,y,10,10)
+        qp.end()
+        """
+        #clickCoorditanes = (x,y) #utworzenie tupli zawierającej wspołrzędne kliknięcia, aby je zwrcić poza funckę
+        
+    def closeEvent(self, event):
+        """
+        funkcja dopytująca czy napewno zamknąć program po kliknięciu na x
+        #!!! Dodać do niej pytanie czy zapisać wyniki do pliku
+        """
+        reply = QMessageBox.question(self, 'Message',
+            "Are you sure to quit?", QMessageBox.Yes | 
+            QMessageBox.No, QMessageBox.No)
+
+        if reply == QMessageBox.Yes:
+            event.accept()
+        else:
+            event.ignore()   
+        
+import shapefile
```

Comment:
Importy dajemy na początku pliku. Tylko jeśli powoduje to błędne koło - robimy to w funkcji.